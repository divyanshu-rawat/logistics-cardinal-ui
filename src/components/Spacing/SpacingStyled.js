import SpacingBlockStyled from './SpacingBlockStyled';

const SpacingStyled = SpacingBlockStyled.extend`
  ${({ type, top, right, bottom, left, theme }) => `
    ${type}-top: ${top ? theme.spacings.kilo : 'auto'};
    ${type}-bottom: ${bottom ? theme.spacings.kilo : 'auto'};
    ${type}-left: ${left ? theme.spacings.kilo : 'auto'};
    ${type}-right: ${right ? theme.spacings.kilo : 'auto'};
  `};
`;

export default SpacingStyled;
