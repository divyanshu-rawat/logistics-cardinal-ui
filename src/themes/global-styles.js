import { createGlobalStyle } from 'styled-components';
import 'normalize.css';

export const GlobalStyles = createGlobalStyle`
  ${({ theme, injectGlobal }) => `
    *,
    *::before,
    *::after {
      box-sizing: border-box;
    }

    html {
      box-sizing: border-box;
      line-height: 1.15;
      -webkit-text-size-adjust: 100%;
      -ms-text-size-adjust: 100%;
      -ms-overflow-style: scrollbar;
      -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
      [type='button'] {
        appearance: none;
      }
    }

    body {
      margin: 0;
      background-color: ${theme.bodyBackground};
      color: ${theme.textColor};
      font-family: ${theme.textFontFamily};
      font-size: ${theme.textFontSize};
      font-weight: ${theme.textFontWeight};
      line-height: ${theme.lineHeightBase};
    }

    fieldset {
      min-width: 0;
      padding: 0;
      margin: 0;
      border: 0;
    }

    .Drawer {
      &--is-open {
        overflow: hidden;
      }
    }

    .ReactModal {
      &__Body--open {
        overflow: hidden;
      }
      &__Overlay {
        position: fixed;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        transition: opacity 200ms ease-in-out;
        height: auto;
        overflow: auto;
        z-index: ${theme.zIndex.modalOverlay};
        background-color: ${theme.overlayColor};
        &--after-open {
          opacity: 1;
        }
        &--before-close {
          opacity: 0;
        }
      }
      &__Content {
        margin: 10em auto;
        position: relative;
        left: auto;
        right: auto;
        top: auto;
        bottom: auto;
        transform: none;
        opacity: 0;
        transition: opacity 200ms ease-in-out;
        min-height: ${theme.modalMinHeight};
        min-width: ${theme.modalMinWidth};
        width: ${theme.modalWidth};
        max-width: ${theme.modalMaxWidth};
        background-color: ${theme.modalBackgroundColor};
        border: 1px ${theme.modalBorderStyle} ${theme.modalBorderColor};
        box-shadow: ${theme.modalBoxShadow};
        border-radius: ${theme.modalBorderRadius};
        outline: ${theme.modalOutline};
        z-index: ${theme.zIndex.modalContent};
        &--after-open {
          opacity: 1;
        }
        &--before-close {
          opacity: 0;
        }
      }
    }


    ${injectGlobal && injectGlobal(theme)};
  `}
`;
