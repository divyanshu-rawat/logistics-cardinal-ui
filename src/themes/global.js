import { injectGlobal } from 'styled-components';

import 'react-select/dist/react-select.css';
import 'react-select-plus/dist/react-select-plus.css';

/* eslint no-unused-expressions: 0 */
injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700,800');

  // used to overwrite bootstrap's background-color
  table[class^="css-"] {
    background: white;
    td[class^="css-"] {
      white-space: initial;
    }
  }
`;
