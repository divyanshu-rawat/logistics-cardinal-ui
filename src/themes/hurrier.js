const white = '#fff';
const black = '#000';

const primary = {
  primary100: '#db0062',
  primary200: '#375f8a',
};

const greys = {
  grey100: '#96a2b0',
  grey200: '#3f454d',
  grey300: '#343a42',
  grey400: '#292d33',
};

const greens = {
  green100: '#c6e5cb',
  green300: '#97e2a3',
  green500: '#62de76',
  green700: '#49b85b',
  green900: '#2f903e',
};

const reds = {
  red100: '#ffe6e1',
  red300: '#ffaf9f',
  red500: '#ff7559',
  red700: '#d55a41',
  red900: '#a73d28',
};

const yellows = {
  yellow100: '#f2e5cb',
  yellow300: '#f7da9f',
  yellow500: '#ffc859',
  yellow700: '#d4a546',
  yellow900: '#ab8433',
};

const misc = {
  bodyBackground: '#fff',
  mainBackground: '#ccc',
  textColor: white,
  danger: reds.red500,
  success: greens.green700,
  warning: yellows.yellow700,
};

const colors = {
  white,
  black,
  ...primary,
  ...greys,
  ...greens,
  ...reds,
  ...yellows,
  ...misc,
};

const spacings = {
  bit: '4px',
  byte: '8px',
  kilo: '12px',
  mega: '16px',
  giga: '24px',
  tera: '32px',
  peta: '40px',
  exa: '48px',
  zetta: '56px',
};

const borderRadius = {
  kilo: '1px',
  mega: '4px',
  giga: '5px',
};

const borderWidth = {
  kilo: '1px',
  mega: '2px',
};

const fontFamily = {
  default: '"Open Sans", sans-serif',
};

const fontWeight = {
  regular: '400',
  bold: '700',
};

export default {
  colors,
  spacings,
  borderRadius,
  borderWidth,
  fontFamily,
  fontWeight,
};
