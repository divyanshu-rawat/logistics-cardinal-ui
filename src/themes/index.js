export rooster from './rooster';
export standard from './standard';
export hurrier from './hurrier';
