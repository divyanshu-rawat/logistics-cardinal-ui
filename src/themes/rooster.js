const white = '#fff';
const black = '#000';

const primary = {
  primary100: '#3f9bfd',
  primary200: '#375f8a',
  primary300: '#324e70',
  primary400: '#96a2b0',
  primary500: '#484e58',
};

const greys = {
  grey100: '#96a2b0',
  grey200: '#3f454d',
  grey300: '#343a42',
  grey400: '#292d33',
  grey500: '#40464f',
  grey600: '#8C9BA5',
  grey700: '#24282E',
  grey800: '#23262C',
  grey900: '#292D33',
  grey1000: '#6d7785',
  grey1100: '#b2b8c0',
  grey1200: '#909295',
  grey1300: '#2c3037',
};

const greens = {
  green100: '#00bc8c',
  green300: '#97e2a3',
  green500: '#62de76',
  green700: '#49b85b',
  green900: '#2f903e',
};

const reds = {
  red100: '#cb3548',
  red300: '#9b5a62',
  red500: '#ff7559',
  red700: '#d55a41',
  red900: '#a94442',
};

const yellows = {
  yellow100: '#d28a2e',
  yellow300: '#f7da9f',
  yellow500: '#ffc859',
  yellow700: '#d4a546',
  yellow900: '#ab8433',
};

const blues = {
  blue100: '#2bdbdb',
};

const colors = {
  white,
  black,
  ...primary,
  ...greys,
  ...greens,
  ...reds,
  ...yellows,
};

const spacings = {
  bit: '1px',
  byte: '5px',
  kilo: '10px',
  mega: '15px',
  giga: '20px',
};

const composedSpacings = {
  byte: `${spacings.kilo} ${spacings.mega}`,
};

const borderRadius = {
  kilo: '1px',

  mega: '4px',
  giga: '5px',
};

const borderWidth = {
  bit: '1px',
  byte: '2px',
};

const fontFamily = {
  default: '"Open Sans", sans-serif',
  color: colors.white,
};

const fontSize = {
  bit: '11px',
  byte: '13px',
  kilo: '15px',
  mega: '18px',
  giga: '24px',
};

const lineHeight = {
  bit: 1.153,
  byte: 1.333,
  kilo: 1.5,
};

const fontWeight = {
  light: 300,
  regular: 400,
  semiBold: 600,
  bold: 700,
};

const boxShadow = {
  switchHandler: '0px 0px 1px 5px rgba(255, 255, 255, 0.2)',
  drawer: '0 2px 34px 0 rgba(0, 0, 0, 0.3)',
  inputButton: 'inset 0 1px 1px rgba(0, 0, 0, 0.075)',
  modalBoxShadow: '0 2px 4px 0 rgba(0, 0, 0, 0.5)',
};

// Required Fields
export const aliases = {
  /**
   * Text Variables
   */
  textColor: white,
  textFontSize: fontSize.byte,
  textFontWeight: fontWeight.regular,
  textFontWeightLight: fontWeight.light,
  textFontFamily: fontFamily.default,
  textSpacing: composedSpacings.byte,
  textFontSizeXSmall: fontSize.bit,
  textFontSizeSmall: fontSize.byte,
  textFontSizeLarge: fontSize.mega,
  textFontSizeXLarge: fontSize.giga,
  /**
   * Heading Variables
   */
  h1FontSize: fontSize.giga,
  h1LineHeight: lineHeight.kilo,
  h2FontSize: fontSize.mega,
  h2LineHeight: lineHeight.byte,
  h3FontSize: fontSize.kilo,
  h3LineHeight: lineHeight.byte,
  h4FontSize: fontSize.byte,
  h4LineHeight: lineHeight.bit,
  h5FontSize: fontSize.bit,
  h5LineHeight: fontSize.bit,
  headingSpacing: spacings.kilo,
  headingFontWeightLight: fontWeight.regular,
  headingFontWeightBold: fontWeight.semiBold,
  /**
   * Color Variables
   */
  bodyBackground: greys.grey400,
  mainBackground: '343941',
  primaryColor: primary.primary100,
  dangerColor: reds.red100,
  successColor: greens.green100,
  warningColor: yellows.yellow700,
  infoColor: blues.blue100,
  overlayColor: 'rgba(25, 29, 35, 0.82)',
  /**
   * Line Height Variables
   */
  lineHeightBase: lineHeight.kilo,
  /**
   * Input and Select Variables
   */
  inputDefaultHeight: '40px',
  // Subtracts border top and bottom
  // It's needed because of react-select-plus
  inputInvisibleHeight: '38px',
  inputPlaceholderColor: greys.grey1200,
  inputBackgroundColor: greys.grey400,
  inputDisabledBackgroundColor: greys.grey300,
  inputBorderColor: greys.grey200,
  inputErrorBorderColor: reds.red300,
  inputBorderColorHover: greys.grey1000,
  inputBorderRadius: borderRadius.mega,
  inputTextColor: white,
  inputLineHeight: '40px',
  inputFontSize: fontSize.byte,
  inputFontWeight: fontWeight.light,
  inputBoxShadow: boxShadow.inputButton,
  inputPadding: composedSpacings.byte,
  // Select
  selectItemFocus: greys.grey300,
  selectItemDisabledColor: greys.grey100,
  selectItemDisabledHoverBackgroundColor: greys.grey400,
  selectMultiTagBackgroundColor: primary.primary100,
  selectMultiTagTextColor: white,
  selectSingleTextColor: white,
  /**
   * Input Feedback Variables
   */
  inputFeedbackSpacing: '0px',
  inputFeedbackFontWeight: fontWeight.semiBold,
  inputFeedbackFontSize: fontSize.byte,
  /**
   * Button Variables
   */
  buttonFontWeight: fontWeight.light,
  buttonFontColor: white,
  buttonBorderWidth: borderWidth.bit,
  buttonFocusBoxShadow: boxShadow.inputButton,
  buttonDisabledOpacity: 0.65,
  buttonActiveBoxShadow: 'inset 0 3px 5px rgba($black, .125)',
  buttonPaddingY: '10px',
  buttonPaddingYXSmall: '1px',
  buttonPaddingYLarge: '18px',
  buttonPaddingYSmall: '6px',
  buttonPaddingSmall: '9px',
  buttonPaddingX: '15px',
  buttonPaddingXSmall: '5px',
  buttonPaddingXLarge: '27px',
  buttonFontSize: fontSize.byte,
  buttonFontSizeSmall: '11px',
  buttonFontSizeLarge: '15px',
  buttonLineHeight: lineHeight.kilo,
  buttonLineHeightSmall: 1.5,
  buttonLineHeightLarge: 1.333,
  buttonBorderRadius: borderRadius.mega,
  /**
   * Radio Button Variables
   */
  radioSpacingRight: spacings.kilo,
  radioFontWeight: fontWeight.regular,
  radioInactiveBorderColor: greys.grey600,
  /**
   * Indicator Radio Button Variables
   */
  indicatorColor: greys.grey200,
  indicatorSpacingRight: spacings.byte,
  indicatorCheckedColor: greys.grey100,
  /**
   * Label Variables
   */
  labelTextColor: primary.primary400,
  labelTextColorError: reds.red100,
  labelFontWeight: fontWeight.regular,
  labelFontSize: fontSize.bit,
  labelBottomSpacing: spacings.byte,
  /**
   * Drawer Variables
   */
  drawerBackgroundColor: greys.grey300,
  drawerBoxShadow: boxShadow.drawer,
  drawerPadding: composedSpacings.byte,
  /**
   * Spacing Variables
   */
  spacing: spacings.kilo,
  /**
   * Spinner Variables
   */
  spinnerOverlayBackgroundColor: 'rgba(52, 58, 66, 0.8)',
  /**
   * Switcher Variables
   */
  switcherSpacingRight: spacings.kilo,
  switcherBorderColor: greys.grey200,
  switcherBoxShadow: boxShadow.switchHandler,
  /**
   * Tabs - Horizontal - Variables
   */
  horizontalTabButtonBorderRadius: borderRadius.mega,
  horizontalTabButtonBackgroundColor: greys.grey300,
  horizontalTabButtonBorderColor: greys.grey400,
  horizontalTabButtonBorderColorActive: greys.grey300,
  horizontalTabButtonColor: greys.grey100,
  horizontalTabButtonColorActive: colors.white,
  horizontalTabButtonSpacingRight: spacings.bit,
  horizontalTabButtonPadding: composedSpacings.byte,
  horizontalTabContainerPadding: composedSpacings.byte,
  horizontalTabContainerBackgroundColor: greys.grey300,
  /**
   * Tabs - Vertical - Variables
   */
  verticalTabButtonBackgroundColor: greys.grey700,
  verticalTabButtonSpacingRight: spacings.bit,
  verticalTabButtonPadding: composedSpacings.byte,
  verticalTabButtonActiveBackgroundColor: greys.grey800,
  verticalTabButtonActiveTextColor: primary.primary100,
  verticalTabButtonTextColor: white,
  verticalTabSidebarBackgroundColor: greys.grey900,
  verticalTabContainerPadding: composedSpacings.byte,
  verticalTabContainerBackgroundColor: greys.grey300,
  /**
   * TimeSpanDisplay - Variables
   */
  timeSpanDisplayTextColor: greys.grey100,
  /**
   * Tooltip - Variables
   */
  tooltipBorderRadius: borderRadius.mega,
  tooltipBackgroundColor: greys.grey400,
  tooltipBorderColor: greys.grey500,
  tooltipTextColor: colors.white,
  tooltipFontSize: fontSize.bit,
  tooltipHoverOpacity: 0.9,
  /**
   * Transitions
   */
  transitions: {
    overlayTransition: 'opacity 0.3s ease-out, visibility 0.3s ease-out',
  },
  /**
   * Modal
   */
  modalBackgroundColor: greys.grey400,
  modalBorderColor: greys.grey200,
  modalBorderStyle: 'solid',
  modalBoxShadow: boxShadow.modalBoxShadow,
  modalBorderRadius: borderRadius.mega,
  modalOutline: 0,
  modalWidth: '80%',
  modalMaxWidth: '60rem',
  modalMinWidth: '20rem',
  minHeight: '10rem',
  /**
   * Accordion
   */
  accordionHeaderHeight: '45px',
  accordionDividerBorderColor: greys.grey1300,
  /**
   * Z-Index
   */
  zIndex: {
    select: 1090,
    tooltip: 2020,
    spinnerOverlay: 2040,
    drawerOverlay: 2040,
    drawerContent: 2041,
    spinner: 2041,
    navbar: 2030,
    loadingBar: 3000,
    modalOverlay: 2050,
    modalContent: 2060,
  },
  /**
   * Media Query Breakpoints
   */
  breakpoints: {
    default: {
      minWidth: 768,
    },
    largeDesktop: {
      minWidth: 1200,
    },
    desktop: {
      minWidth: 992,
      maxWidth: 1200,
    },
    mobile: {
      minWidth: 0,
      maxWidth: 768,
    },
    tablet: {
      minWidth: 768,
      maxWidth: 992,
    },
  },
  grid: {
    outerGutter: '10px',
  },
  colors,
  fontSize,
  fontWeight,
};

export const RoosterTheme = {
  ...aliases,
};
