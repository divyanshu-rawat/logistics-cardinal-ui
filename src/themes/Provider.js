import React, { Fragment } from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { standard } from '.';
import { GlobalStyles } from './global-styles';

const CardinalThemeProvider = ({ theme, injectGlobal, children }) => {
  const mergedTheme = {
    ...standard,
    ...theme,
  };

  return (
    <ThemeProvider theme={mergedTheme}>
      <Fragment>
        <GlobalStyles injectGlobal={injectGlobal} />
        {children}
      </Fragment>
    </ThemeProvider>
  );
};

CardinalThemeProvider.propTypes = {
  children: PropTypes.node.isRequired,
  injectGlobal: PropTypes.func,
  theme: PropTypes.object,
};

export { CardinalThemeProvider };
