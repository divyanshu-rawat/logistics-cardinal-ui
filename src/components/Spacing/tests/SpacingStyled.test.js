import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import SpacingStyled from '../SpacingStyled';

import { rooster as RoosterTheme } from '../../../themes';

const renderedComponent = ({ type, top, bottom, right, left }) =>
  mount(
    <SpacingStyled
      type={type}
      top={top}
      right={right}
      bottom={bottom}
      left={left}
      theme={RoosterTheme}
    />,
  );

describe('<SpacingStyled />', () => {
  it('should render the component with the default props', () => {
    const tree = toJson(renderedComponent({ type: 'margin' }));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with padding instead of margin', () => {
    const tree = toJson(
      renderedComponent({
        type: 'padding',
        top: true,
        bottom: true,
        right: true,
        left: true,
      }),
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with all properties as true', () => {
    const tree = toJson(
      renderedComponent({
        type: 'margin',
        top: true,
        bottom: true,
        right: true,
        left: true,
      }),
    );
    expect(tree).toMatchSnapshot();
  });
});
