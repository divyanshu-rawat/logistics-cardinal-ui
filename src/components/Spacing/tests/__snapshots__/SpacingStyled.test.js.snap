// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpacingStyled /> should render the component with all properties as true 1`] = `
<SpacingBlockStyled
  bottom={true}
  left={true}
  right={true}
  theme={
    Object {
      "borderRadius": Object {
        "giga": "5px",
        "kilo": "1px",
        "mega": "4px",
      },
      "borderWidth": Object {
        "bit": "1px",
        "byte": "2px",
      },
      "boxShadow": Object {
        "drawer": "0 2px 34px 0 rgba(0, 0, 0, 0.3)",
        "switchHandler": "0px 0px 1px 5px rgba(255, 255, 255, 0.2)",
      },
      "breakpoints": Object {
        "default": Object {
          "minWidth": 768,
        },
        "desktop": Object {
          "maxWidth": 1200,
          "minWidth": 992,
        },
        "largeDesktop": Object {
          "minWidth": 1200,
        },
        "mobile": Object {
          "maxWidth": 768,
          "minWidth": 0,
        },
        "tablet": Object {
          "maxWidth": 992,
          "minWidth": 768,
        },
      },
      "colors": Object {
        "black": "#000",
        "bodyBackground": "#292d33",
        "danger": "#a94442",
        "green100": "#c6e5cb",
        "green300": "#97e2a3",
        "green500": "#62de76",
        "green700": "#49b85b",
        "green900": "#2f903e",
        "grey100": "#96a2b0",
        "grey1000": "#6d7785",
        "grey1100": "#b2b8c0",
        "grey200": "#3f454d",
        "grey300": "#343a42",
        "grey400": "#292d33",
        "grey500": "#40464f",
        "grey600": "#8C9BA5",
        "grey700": "#24282E",
        "grey800": "#23262C",
        "grey900": "#292D33",
        "mainBackground": "343941",
        "overlayColor": "rgba(25, 29, 35, 0.82)",
        "primary100": "#3f9bfd",
        "primary200": "#375f8a",
        "primary300": "#324e70",
        "primary400": "#96a2b0",
        "primary500": "#484e58",
        "red100": "#cb3548",
        "red300": "#9b5a62",
        "red500": "#ff7559",
        "red700": "#d55a41",
        "red900": "#a94442",
        "success": "#49b85b",
        "textColor": "#fff",
        "warning": "#d4a546",
        "white": "#fff",
        "yellow100": "#f2e5cb",
        "yellow300": "#f7da9f",
        "yellow500": "#ffc859",
        "yellow700": "#d4a546",
        "yellow900": "#ab8433",
      },
      "composedSpacings": Object {
        "byte": "10px 15px",
      },
      "fontFamily": Object {
        "color": "#fff",
        "default": "\\"Open Sans\\", sans-serif",
      },
      "fontSize": Object {
        "bit": "11px",
        "byte": "13px",
        "giga": "24px",
        "kilo": "15px",
        "mega": "18px",
      },
      "fontWeight": Object {
        "bold": 700,
        "light": 300,
        "regular": 400,
        "semiBold": 600,
      },
      "grid": Object {
        "outerGutter": "10px",
      },
      "lineHeight": Object {
        "bit": 1.153,
        "byte": 1.333,
        "kilo": 1.5,
      },
      "spacings": Object {
        "bit": "1px",
        "byte": "5px",
        "giga": "20px",
        "kilo": "10px",
        "mega": "15px",
      },
      "transitions": Object {
        "overlayTransition": "opacity 0.3s ease-out, visibility 0.3s ease-out",
      },
      "zIndex": Object {
        "drawerContent": 2041,
        "drawerOverlay": 2040,
        "navbar": 2030,
        "select": 1090,
        "spinner": 2010,
        "spinnerOverlay": 2040,
        "tooltip": 2020,
      },
    }
  }
  top={true}
  type="margin"
>
  <div
    className="SpacingBlockStyled-cDrbtl eoxfIb"
    type="margin"
  />
</SpacingBlockStyled>
`;

exports[`<SpacingStyled /> should render the component with padding instead of margin 1`] = `
<SpacingBlockStyled
  bottom={true}
  left={true}
  right={true}
  theme={
    Object {
      "borderRadius": Object {
        "giga": "5px",
        "kilo": "1px",
        "mega": "4px",
      },
      "borderWidth": Object {
        "bit": "1px",
        "byte": "2px",
      },
      "boxShadow": Object {
        "drawer": "0 2px 34px 0 rgba(0, 0, 0, 0.3)",
        "switchHandler": "0px 0px 1px 5px rgba(255, 255, 255, 0.2)",
      },
      "breakpoints": Object {
        "default": Object {
          "minWidth": 768,
        },
        "desktop": Object {
          "maxWidth": 1200,
          "minWidth": 992,
        },
        "largeDesktop": Object {
          "minWidth": 1200,
        },
        "mobile": Object {
          "maxWidth": 768,
          "minWidth": 0,
        },
        "tablet": Object {
          "maxWidth": 992,
          "minWidth": 768,
        },
      },
      "colors": Object {
        "black": "#000",
        "bodyBackground": "#292d33",
        "danger": "#a94442",
        "green100": "#c6e5cb",
        "green300": "#97e2a3",
        "green500": "#62de76",
        "green700": "#49b85b",
        "green900": "#2f903e",
        "grey100": "#96a2b0",
        "grey1000": "#6d7785",
        "grey1100": "#b2b8c0",
        "grey200": "#3f454d",
        "grey300": "#343a42",
        "grey400": "#292d33",
        "grey500": "#40464f",
        "grey600": "#8C9BA5",
        "grey700": "#24282E",
        "grey800": "#23262C",
        "grey900": "#292D33",
        "mainBackground": "343941",
        "overlayColor": "rgba(25, 29, 35, 0.82)",
        "primary100": "#3f9bfd",
        "primary200": "#375f8a",
        "primary300": "#324e70",
        "primary400": "#96a2b0",
        "primary500": "#484e58",
        "red100": "#cb3548",
        "red300": "#9b5a62",
        "red500": "#ff7559",
        "red700": "#d55a41",
        "red900": "#a94442",
        "success": "#49b85b",
        "textColor": "#fff",
        "warning": "#d4a546",
        "white": "#fff",
        "yellow100": "#f2e5cb",
        "yellow300": "#f7da9f",
        "yellow500": "#ffc859",
        "yellow700": "#d4a546",
        "yellow900": "#ab8433",
      },
      "composedSpacings": Object {
        "byte": "10px 15px",
      },
      "fontFamily": Object {
        "color": "#fff",
        "default": "\\"Open Sans\\", sans-serif",
      },
      "fontSize": Object {
        "bit": "11px",
        "byte": "13px",
        "giga": "24px",
        "kilo": "15px",
        "mega": "18px",
      },
      "fontWeight": Object {
        "bold": 700,
        "light": 300,
        "regular": 400,
        "semiBold": 600,
      },
      "grid": Object {
        "outerGutter": "10px",
      },
      "lineHeight": Object {
        "bit": 1.153,
        "byte": 1.333,
        "kilo": 1.5,
      },
      "spacings": Object {
        "bit": "1px",
        "byte": "5px",
        "giga": "20px",
        "kilo": "10px",
        "mega": "15px",
      },
      "transitions": Object {
        "overlayTransition": "opacity 0.3s ease-out, visibility 0.3s ease-out",
      },
      "zIndex": Object {
        "drawerContent": 2041,
        "drawerOverlay": 2040,
        "navbar": 2030,
        "select": 1090,
        "spinner": 2010,
        "spinnerOverlay": 2040,
        "tooltip": 2020,
      },
    }
  }
  top={true}
  type="padding"
>
  <div
    className="SpacingBlockStyled-cDrbtl csVoMA"
    type="padding"
  />
</SpacingBlockStyled>
`;

exports[`<SpacingStyled /> should render the component with the default props 1`] = `
<SpacingBlockStyled
  theme={
    Object {
      "borderRadius": Object {
        "giga": "5px",
        "kilo": "1px",
        "mega": "4px",
      },
      "borderWidth": Object {
        "bit": "1px",
        "byte": "2px",
      },
      "boxShadow": Object {
        "drawer": "0 2px 34px 0 rgba(0, 0, 0, 0.3)",
        "switchHandler": "0px 0px 1px 5px rgba(255, 255, 255, 0.2)",
      },
      "breakpoints": Object {
        "default": Object {
          "minWidth": 768,
        },
        "desktop": Object {
          "maxWidth": 1200,
          "minWidth": 992,
        },
        "largeDesktop": Object {
          "minWidth": 1200,
        },
        "mobile": Object {
          "maxWidth": 768,
          "minWidth": 0,
        },
        "tablet": Object {
          "maxWidth": 992,
          "minWidth": 768,
        },
      },
      "colors": Object {
        "black": "#000",
        "bodyBackground": "#292d33",
        "danger": "#a94442",
        "green100": "#c6e5cb",
        "green300": "#97e2a3",
        "green500": "#62de76",
        "green700": "#49b85b",
        "green900": "#2f903e",
        "grey100": "#96a2b0",
        "grey1000": "#6d7785",
        "grey1100": "#b2b8c0",
        "grey200": "#3f454d",
        "grey300": "#343a42",
        "grey400": "#292d33",
        "grey500": "#40464f",
        "grey600": "#8C9BA5",
        "grey700": "#24282E",
        "grey800": "#23262C",
        "grey900": "#292D33",
        "mainBackground": "343941",
        "overlayColor": "rgba(25, 29, 35, 0.82)",
        "primary100": "#3f9bfd",
        "primary200": "#375f8a",
        "primary300": "#324e70",
        "primary400": "#96a2b0",
        "primary500": "#484e58",
        "red100": "#cb3548",
        "red300": "#9b5a62",
        "red500": "#ff7559",
        "red700": "#d55a41",
        "red900": "#a94442",
        "success": "#49b85b",
        "textColor": "#fff",
        "warning": "#d4a546",
        "white": "#fff",
        "yellow100": "#f2e5cb",
        "yellow300": "#f7da9f",
        "yellow500": "#ffc859",
        "yellow700": "#d4a546",
        "yellow900": "#ab8433",
      },
      "composedSpacings": Object {
        "byte": "10px 15px",
      },
      "fontFamily": Object {
        "color": "#fff",
        "default": "\\"Open Sans\\", sans-serif",
      },
      "fontSize": Object {
        "bit": "11px",
        "byte": "13px",
        "giga": "24px",
        "kilo": "15px",
        "mega": "18px",
      },
      "fontWeight": Object {
        "bold": 700,
        "light": 300,
        "regular": 400,
        "semiBold": 600,
      },
      "grid": Object {
        "outerGutter": "10px",
      },
      "lineHeight": Object {
        "bit": 1.153,
        "byte": 1.333,
        "kilo": 1.5,
      },
      "spacings": Object {
        "bit": "1px",
        "byte": "5px",
        "giga": "20px",
        "kilo": "10px",
        "mega": "15px",
      },
      "transitions": Object {
        "overlayTransition": "opacity 0.3s ease-out, visibility 0.3s ease-out",
      },
      "zIndex": Object {
        "drawerContent": 2041,
        "drawerOverlay": 2040,
        "navbar": 2030,
        "select": 1090,
        "spinner": 2010,
        "spinnerOverlay": 2040,
        "tooltip": 2020,
      },
    }
  }
  type="margin"
>
  <div
    className="SpacingBlockStyled-cDrbtl eIkgod"
    type="margin"
  />
</SpacingBlockStyled>
`;
